paperqa/agents/helpers.py

```diff
@@ -0,0 +1,92 @@
+from __future__ import annotations
+
+import logging
+import re
+from typing import cast
+
+from openai import AsyncOpenAI
+from rich.table import Table
+
+from .. import OpenAILLMModel
+from ..docs import Docs
+from .models import AnswerResponse
+from .tools import get_year
+
+logger = logging.getLogger(__name__)
+
+
+async def openai_get_search_query(
+    question: str,
+    count: int,
+    template: str | None = None,
+    llm: str = "gpt-4o-mini",
+    temperature: float = 1.0,
+) -> list[str]:
+    if isinstance(template, str):
+        if not (
+            "{count}" in template and "{question}" in template and "{date}" in template
+        ):
+            logger.warning(
+                "Template does not contain {count}, {question} and {date} variables. Ignoring template."
+            )
+            template = None
+
+        else:
+            # partial formatting
+            search_prompt = template.replace(r"{date}", get_year())
+
+    if template is None:
+        search_prompt = (
+            "We want to answer the following question: {question} \n"
+            "Provide {count} unique keyword searches (one search per line) and year ranges "
+            "that will find papers to help answer the question. "
+            "Do not use boolean operators. "
+            "Make sure not to repeat searches without changing the keywords or year ranges. "
+            "Make some searches broad and some narrow. "
+            "Use this format: [keyword search], [start year]-[end year]. "
+            "where end year is optional. "
+            f"The current year is {get_year()}\n\n"
+        )
+
+    if "gpt" not in llm:
+        raise ValueError(
+            f"Invalid llm: {llm}, note a GPT model must be used for the fake agent search."
+        )
+    client = AsyncOpenAI()
+    model = OpenAILLMModel(config={"model": llm, "temperature": temperature})
+    chain = model.make_chain(client, prompt=search_prompt, skip_system=True)
+    result = await chain({"question": question, "count": count})  # type: ignore[call-arg]
+    search_query = result.text
+    queries = [s for s in search_query.split("\n") if len(s) > 3]  # noqa: PLR2004
+    # remove 2., 3. from queries
```

Comment:
What is `2., 3.`? I am not quite following this comment

Reply:
it's literal removing "2., 3." for example from a query i added a quote

Reply:
https://regex101.com/r/W2f7F1/1